### ISSUE TITLE
Data loading error on Table sorting leads to NPE in vaadin code

### ISSUE THREAD

**Issue by alexbudarov** [cuba-platform/cuba#2138]
### Environment
- Platform version: 6.10

### Description of the bug or enhancement
1) Use refapp 6.10.
Apply attached patch:
[emulate-data-loading-error.patch.txt](https://github.com/cuba-platform/cuba/files/3202285/emulate-data-loading-error.patch.txt)

It needs to emulate data loading error - DevelopmentException which is processed by exception handlers.
Any query having "DriverCallsign" will fail with error. Any query with "order by u.email" will fail with error.
In real case it was an sql query timeout exception with its own exception handler.

2) Start server.
3) Go to Application -> Driver Callsigns.
You see a nice shiny dialog, as expected:
<details>
 <summary>Click to expand</summary>
 
![image](https://user-images.githubusercontent.com/6158807/58092952-6a6d1280-7bde-11e9-8cb1-5cfc447c5a0c.png)
</details>

4) Insert more than 50 sec$User instances to database. It is necessary to enable database-level table sorting. E.g. this query:
<details>
 <summary>Click to expand</summary>
 
```
insert into sec_user (id, version, create_ts, created_by, update_ts, login, login_lc, name, email, group_id, position_)
select newid(), 1, now(), 'admin', now(), 
'us' ||  t, 'us' ||  t, 
'Name ' || t, 'user' || t || '@example.com', 
'0fa2b1a5-1d68-4d69-9fbd-dff348347f93', 'Manager'
from generate_series(1,1000) t
```
</details>

5) Open Administration -> Users.
It opens successfully - there is no sorting.

6) Click "Email" column header. It triggers database sorting.
You get an "Internal error" and this exception in log:

<details>
 <summary>Click to expand</summary>
 
```
2019-05-21 15:41:04.683 ERROR [http-nio-8080-exec-8/app/admin] com.haulmont.cuba.web.log.AppLog - Exception: 
java.lang.RuntimeException: Push failed
	at com.vaadin.server.communication.AtmospherePushConnection.push(AtmospherePushConnection.java:177) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.server.communication.AtmospherePushConnection.push(AtmospherePushConnection.java:152) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.ui.UI.push(UI.java:1562) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.server.VaadinSession.unlock(VaadinSession.java:1004) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:43) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1435) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.server.VaadinServlet.service(VaadinServlet.java:361) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.haulmont.cuba.web.sys.CubaApplicationServlet.serviceAppRequest(CubaApplicationServlet.java:324) ~[cuba-web-6.10-SNAPSHOT.jar:6.10-SNAPSHOT]
	at com.haulmont.cuba.web.sys.CubaApplicationServlet.service(CubaApplicationServlet.java:210) ~[cuba-web-6.10-SNAPSHOT.jar:6.10-SNAPSHOT]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:742) ~[servlet-api.jar:na]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) ~[catalina.jar:8.5.33]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.33]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) ~[tomcat-websocket.jar:8.5.33]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.33]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.33]
	at org.springframework.web.filter.CompositeFilter$VirtualFilterChain.doFilter(CompositeFilter.java:107) ~[spring-web-4.3.23.RELEASE.jar:4.3.23.RELEASE]
	at org.springframework.web.filter.CompositeFilter.doFilter(CompositeFilter.java:73) ~[spring-web-4.3.23.RELEASE.jar:4.3.23.RELEASE]
	at com.haulmont.cuba.web.sys.CubaHttpFilter.doFilter(CubaHttpFilter.java:107) ~[cuba-web-6.10-SNAPSHOT.jar:6.10-SNAPSHOT]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) ~[catalina.jar:8.5.33]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) ~[catalina.jar:8.5.33]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198) ~[catalina.jar:8.5.33]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) ~[catalina.jar:8.5.33]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:493) ~[catalina.jar:8.5.33]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140) ~[catalina.jar:8.5.33]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) ~[catalina.jar:8.5.33]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:650) ~[catalina.jar:8.5.33]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) ~[catalina.jar:8.5.33]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342) ~[catalina.jar:8.5.33]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:800) ~[tomcat-coyote.jar:8.5.33]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66) ~[tomcat-coyote.jar:8.5.33]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:806) ~[tomcat-coyote.jar:8.5.33]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1498) ~[tomcat-coyote.jar:8.5.33]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-coyote.jar:8.5.33]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_201]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_201]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-util.jar:8.5.33]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_201]
Caused by: java.lang.NullPointerException: null
	at com.vaadin.ui.Table.findNumRowsToPaint(Table.java:3826) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.ui.Table.doPaintContent(Table.java:3324) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.ui.Table.paintContent(Table.java:3310) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.haulmont.cuba.web.toolkit.ui.CubaTable.paintContent(CubaTable.java:889) ~[cuba-web-6.10-SNAPSHOT.jar:6.10-SNAPSHOT]
	at com.haulmont.cuba.web.toolkit.ui.CubaGroupTable.paintContent(CubaGroupTable.java:102) ~[cuba-web-6.10-SNAPSHOT.jar:6.10-SNAPSHOT]
	at com.vaadin.server.LegacyPaint.paint(LegacyPaint.java:65) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.server.communication.LegacyUidlWriter.write(LegacyUidlWriter.java:82) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.server.communication.UidlWriter.write(UidlWriter.java:131) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	at com.vaadin.server.communication.AtmospherePushConnection.push(AtmospherePushConnection.java:174) ~[vaadin-server-7.7.15.cuba.10.jar:7.7.15.cuba.10]
	... 36 common frames omitted
```
![image](https://user-images.githubusercontent.com/6158807/58093367-6beb0a80-7bdf-11e9-98c5-a791671bca1b.png)

</details>

Expected: the same nice "Unexpected error" dialog.

**Comment by web-devel** [comment]
Test project
[table-sorting-npe.zip](https://github.com/cuba-platform/cuba/files/3358621/table-sorting-npe.zip)
* Open Main Menu > Users screen
* Press generate 100 users button
* Toggle sort on some column > See development exception dialog
* Press Reload

AR:
Internal error message
ER:
Table data reloaded

**Comment by bapehbe** [comment]
Is there a chance to get it fixed in 6.10?
